### ISSUE TITLE
JAWS announces entire contents of dialog the first time a button is focused

### ISSUE THREAD

**Issue by jreid01** [FreedomScientific/VFO-standards-support#662]
Steps to reproduce:
1. Go to https://jsfiddle.net/jreid01/0zcjfoyw/2/
2. Activate JAWS
3. Focus the Name input field
4. Tab to the cancel button.

Expected result:
JAWS should announce "Cancel button."

Actual result:
JAWS announces entire contents of dialog, "Screen reader dialog test case name new cuecard first choice second choice," before announcing the name of the cancel button. This seems to happen only the first time a button is focused; the problem does not happen when focusing subsequent buttons, or leaving the first button and returning to it.

Note: This also happens when JAWS reads the dialog example on the Modal Dialog Example page on the APG. Steps to reproduce there:
1. Go to https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html
2. Tab to Add Delivery Address button in the example. Activate the button.
3. Tab to the Verify Address button.

Expected result:
JAWS should announce "Verify Address."

Actual result:
JAWS announces entire contents of dialog before announcing the button.

JAWS version 2022 2204.20 ILM + Chrome 104.0.5112.102 (64 bit win)

**Comment by stevefaulkner** [comment]
Works as expected in latest JAWS 2023+Chrome
[test case](https://codepen.io/stevef/pen/dygBgwp) 

**Note** 
- I provided an accessible name for the `dialog` (which is a requirement, otherwise the screen reader may well use part or all of the text in the `dialog` as an acc name) 
-  i added labels to the radio buttons rather than dupe the visible text with `aria-label`
- removed _button_ from acc name of button's as the role is announced anyway

```
<div role="dialog" aria-labelledby="heading1">
  <h1 id="heading1">Screen Reader Dialog Test Case</h1>
  <label>Name  <input type="text" value="New cuecard"></label>
  <div>
    <label><input type="radio" checked name="test-radio" value="First choice"> First choice</label>
    <label><input type="radio" name="test-radio" value="Second choice"> Second choice</label>
  </div>
  <div>
    <button>cancel</button>
    <button>ok</button>
  </div>
</div>
```

**Comment by JianMeng** [comment]
@stevefaulkner  I meet same issue. But in my case, not full content of dialog read, only half of it is read. Do you know why jaws add this feature?

**Comment by stevefaulkner** [comment]
> @stevefaulkner I meet same issue. But in my case, not full content of dialog read, only half of it is read. Do you know why jaws add this feature?

@JianMeng i just rechecked the [APG example](https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html) with JAWS 2023.2306.38 and Chrome Version 115.0.5790.110 (Official Build) (64-bit). It works as expected. Unless you can provide a test case where you are experiencing this issue, then it's not possible to answer your question.

**Comment by ramjiseetharaman2425** [comment]
@stevefaulkner - Able to reproduce this issue in the ARIA APG examples for Modal dialogs. I am using Google Chrome build # 119.6045.160 and JAWS 2023 version 2023.2302.15. 

Attaching the JAWS speech history for your reference.
![IMG_0009](https://github.com/FreedomScientific/standards-support/assets/156229561/c4ca6164-80d8-4294-b8e0-3932e32d86d4)

Alt Text: Screenshot of the ARIA apg Dialog pattern example along with the JAWS speech history for the dialog when modal is open.

**Comment by aardrian** [comment]
Using Chrome 120 with JAWS 2024.2312.53, when I open any of the three examples and tab to any button within the dialog after opening the dialog I do not experience what you report @ramjiseetharaman2425 .

The screen shot you are sharing shows me that you activated the "Add Delivery Address" button (APG example), which then triggers the dialog. Then it reads all the field names and then puts focus on the Street field. The original issue was filed as reading the entire dialog when focus is put on a button _within_ the dialog.

Does that behavior happen with the other two examples?

When I activate the dialog, I only hear the Street field, which is what gets focus.

When you open the dialog, does focus immediately go to the first field or is there some delay?

**Comment by ramjiseetharaman2425** [comment]
@aardrian - Thank you so much for your response. Yes, validations were performed with Date Picker dialog, Alert Dialog as well as the Modal dialog example

For date picker dialog, when the dialog initially opens when you activate the calendar button, JAWS renders Choose Date Dialog, January 2024, Cursor keys can navigate dates, 11 button.

For the Add delivery address dialog, whenever it opens focus is automatically set on the Street input field. I don't foresee any delays with the focus being said.

When the dialog is open, it identifies all static text and renders it before it gives information of the element that is currently focused. So, it reads Add Delivery address dialog, all input field labels and then the special instructions hint text and then gives information about the Street address input!

**NOTE:** Adding aria-describedby attribute on the modal dialog container and pointing it to an empty div with a non blocking space seems to not do that!

Browser: Chrome 119.0.6045.160 Official Build
JAWS: 2023.2302.15
